functions.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+
+/**
+ * connects SQL database to PHP
+ */
+function connectDB()
+{
+    $db = new PDO ('mysql:host=db; dbname=AlexCollection', 'root', 'password');
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
+}
+
+/**
+ * pulls set names from PDO
+ *
+ * @param array $db the PDO connection to SQL
+ *
+ * @return string the data from SQL
+ */
+function pullSetData(PDO $db) :array
+{
+    $query = $db->prepare("SELECT `name`, `released`, `cards` FROM `MTGSets`");
+    $query->execute();
+    $result = $query-> fetchAll();
+    $sets = $result;
+    return $sets;
+}
+
+/**
+ * displays set data in divs
+ *
+ * @param array $sets the set data from SQL
```

Comment:
try to make description as generic as poss, this function could be reused regardless of the data source, as long as in the correct format. eg: an indexed array of assoc arrays of card sets